src/controllers/getRecipeByIdController.ts

```diff
@@ -0,0 +1,61 @@
+import { Request, Response } from "express";
+import getDatabase from "../Services/databaseConnector";
+import { userIdExists, recipeIdExists } from "../Services/validators";
+import { Connection } from "promise-mysql";
+
+interface Ingredient {id: number, name: string}
+interface Recipe {
+  id: number,
+  name: string,
+  instructions: string,
+  prep_time: number,
+  cook_time: number,
+  ingredients: [Ingredient]
+}
+
+export async function getRecipeById(
+  req: Request,
+  res: Response
+): Promise<void> {
+  try {
+    const db: Connection = await getDatabase();
+    const userId: number = Number(req.params.userId);
+    const recipeId: number = Number(req.params.recipeId);
```

Comment:
Neither of these IDs are actually validated. You must validate your IDs.

Reply:
done